### ISSUE TITLE
C/C++: can not parse file named "ld.h" or "ld.c"

### ISSUE THREAD

**Issue by chongchai** [universal-ctags/ctags#2603]
I find the ctags can't parse "ld.h" file. Is there any limits on file name?

My test follows: 

**ld.h**：
```
void test();
```
**command**:
```ctags.exe --output-format=json --langmap=c:+.inc --kinds-c=+plz --fields=+nie -o - D:\ld.h```

**result**
it gets empty result.

If rename the file `ld.h` to other name like `a.h`, I will get correct result.

**Comment by masatake** [comment]
Thank you for reporting.

Please, try running ctags with adding  `--map-LdScript=-'(ld.*)'` to the command line as a workaround.
```
$ ./ctags --list-maps=LdScript
LdScript *.lds.S ld.* *.lds *.scr *.ld *.ldi
```
It seems that the definition of the filename pattern map for LdScript parser is too aggressive.
I will make a pull request for weakening the definition.

**Comment by chongchai** [comment]
Adding `--map-LdScript=-"(ld.*)"` can solve my problem. Thank you!

**Comment by masatake** [comment]
Fixed in #2605.
